crates/wast/src/wast.rs

```diff
@@ -30,8 +30,8 @@ pub struct WastContext {
     current: Option<Instance>,
 
     instances: HashMap<String, Instance>,
+    linker: Linker,
```

Comment:
It feels awkward to have both `instances` and `linker`, since they're largely overlapping. Code like `fn invoke` currently looks up a method to invoke by doing `get_instance` and then looking up the export. Can it user the `linker` to do this instead? It seems like we should be able to completely eliminate `instances`.

Reply:
I agree yeah it's a bit awkward but I don't think we can easily change this. This would involve adding some sort of accesor to `Linker` and it wouldn't work well for the `register` command which takes a whole instance and puts it in the linker. Would you be ok trying to clean this up at a later date?

Reply:
Yeah. In fact, I think I have an idea of how to do it, so let's go ahead and land this and I'll see if my idea works in a separate PR.

Reply:
I took a quick look at this, and it turns out to be harder than I thought. The Linker interface doesn't have a way to look up a function without knowing its return types, which is something that the wast `invoke` command does. We may eventually want to add a way to do this, but we can figure that out later.